test/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter_test.cc

```diff
@@ -55,6 +76,105 @@ class AdaptiveConcurrencyFilterTest : public testing::Test {
   std::unique_ptr<AdaptiveConcurrencyFilter> filter_;
 };
 
+TEST_F(AdaptiveConcurrencyFilterTest, TestEnableOverriddenFromRuntime) {
+  std::string yaml_config =
+      R"EOF(
+gradient_controller_config:
+  sample_aggregate_percentile:
+    value: 50
+  concurrency_limit_params:
+    concurrency_update_interval:
+      nanos: 100000000 # 100ms
+  min_rtt_calc_params:
+    interval:
+      seconds: 30
+    request_count: 50
+enabled:
+  default_value: true
+  runtime_key: "adaptive_concurrency.enabled"
+)EOF";
+
+  auto config = makeConfig(yaml_config);
+
+  auto config_ptr = std::make_shared<AdaptiveConcurrencyFilterConfig>(
+      config, runtime_, "testprefix.", stats_, time_system_);
+  filter_ = std::make_unique<AdaptiveConcurrencyFilter>(config_ptr, controller_);
+  filter_->setDecoderFilterCallbacks(decoder_callbacks_);
+  filter_->setEncoderFilterCallbacks(encoder_callbacks_);
+
+  // The filter should behave as normal here.
+
+  Http::TestHeaderMapImpl request_headers;
+
+  EXPECT_CALL(*controller_, forwardingDecision())
+      .WillOnce(Return(RequestForwardingAction::Forward));
+  EXPECT_CALL(*controller_, recordLatencySample(_));
+
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+
+  Buffer::OwnedImpl request_body;
+  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_->decodeData(request_body, false));
+
+  Http::TestHeaderMapImpl request_trailers;
+  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_->decodeTrailers(request_trailers));
+
+  Http::TestHeaderMapImpl response_headers;
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->encodeHeaders(response_headers, false));
+  filter_->encodeComplete();
+
+  // Request should be blocked when flag is overridden. Note there is no expected call to
```

Comment:
nit: "The filter will be disabled when the flag is overridden"

Also, as a minor issue, please put this part in a dedicated test so we better mimic production in which a filter is only used once per request/response. Please fix this in other tests where applicable.